src/compiler/checker.ts

```diff
@@ -28515,11 +28515,45 @@ namespace ts {
             }
         }
 
+        function getNonSimpleParameters(parameters: ReadonlyArray<ParameterDeclaration>): ReadonlyArray<ParameterDeclaration> {
+            // ECMA-262 14.1.13
+            if (parameters.length === 0) return [];
+
+            const last = lastOrUndefined(parameters);
+            if (last && isRestParameter(last)) return [last];
+
+            return filter(parameters, parameter => {
+                // ECMA-262 13.3.3.4
+                return !!parameter.initializer || isBindingPattern(parameter.name);
+            });
+        }
+
+        function checkGrammarForUseStrictSimpleParameterList (node: FunctionLikeDeclaration): boolean {
+            const useStrictDirective = node.body && isBlock(node.body) && startsWithUseStrict(node.body.statements) ? first(node.body.statements) : undefined;

```

Comment:
`startsWithUseStrict` is actually not correct. It doesn't handle the case where `'use strict'` is not the first directive in the prologue:

```ts
function foo(param = 1) {
  'foo';
  'use strict'; // this is also a syntax error
}
```

Also note that `startsWithUseStrict` is also not correct about what it considers a Use Strict Directive:
`'use\u0020strict';` is not a valid Use Strict Directive, but the function thinks it is.
Though that is probably not your responsibility to fix it in this PR.